### PR TITLE
Refactor/cmake

### PR DESCRIPTION
**Pull request by nidegen**
Refactored Cmake for easy dependency management

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,72 +15,85 @@ IF(NOT CMAKE_BUILD_TYPE)
...
-   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-   ADD_DEFINITIONS(-DCOMPILEDWITHC11)
-   MESSAGE(STATUS "Using flag -std=c++11.")
-ELSE()
-   MESSAGE(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
-ENDIF()
+SET(CMAKE_CXX_STANDARD 11)
+SET(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# TODO(nidgen) discuss/check if following is needed: (from original CMakeLists.txt)
```
Comment (gocarlos) [inline CMakeLists.txt]: We could then also remove some code from the ORB_SLAM2 src, they had some ifdefs for the case where not c++11 was available.

Comment (nidegen) [reply CMakeLists.txt]: ok. i think we can assume C++11 is given, right?

Comment (gocarlos) [reply CMakeLists.txt]: yes, I means every compiler has c++11 today,

File: CMakeLists.txt [hunk 2]
```diff
@@ -10,72 +15,85 @@ IF(NOT CMAKE_BUILD_TYPE)
...
-  ${CMAKE_CURRENT_LIST_DIR}/src/Sim3Solver.cc
-  ${CMAKE_CURRENT_LIST_DIR}/src/Initializer.cc
-  ${CMAKE_CURRENT_LIST_DIR}/src/Viewer.cc
+TARGET_LINK_LIBRARIES(${ORB_SLAM2_LIB}
+  debug ${OpenCV_LIBRARIES} optimized ${OpenCV_LIBRARIES}
+  debug ${DBOW2_LIB} optimized ${DBOW2_LIB}
+  debug ${G2O_LIB} debug ${G2O_LIB}
 )
 
-IF(APPLE)
```
Comment (gocarlos) [inline CMakeLists.txt:67]: :+1:

File: Thirdparty/g2o/CMakeLists.txt [hunk 3]
```diff
@@ -1,174 +1,76 @@
...
-# Detect OS and define macros appropriately
-IF(UNIX)
-  ADD_DEFINITIONS(-DUNIX)
-  MESSAGE(STATUS "Compiling on Unix")
-ENDIF(UNIX)
-
-# Eigen library parallelise itself, though, presumably due to performance issues
-# OPENMP is experimental. We experienced some slowdown with it
-FIND_PACKAGE(OpenMP)
-SET(G2O_USE_OPENMP OFF CACHE BOOL "Build g2o with OpenMP support (EXPERIMENTAL)")
```
Comment (gocarlos) [inline Thirdparty/g2o/CMakeLists.txt:48]: No need for OpenMP anymore?

Comment (nidegen) [reply Thirdparty/g2o/CMakeLists.txt:48]: I think it was off by default, and i never had it on mac.

C++11 may cover the multithreading stuff mabe? didt go into details

Comment (gocarlos) [reply Thirdparty/g2o/CMakeLists.txt:48]: aha, I didn't see that it was off by default.. 
okay

**Comment by gocarlos** [comment]
regarding the eigen3 library, do you think we could use the cmake-module to find eigen3 on travis? 
Or is there another option to find eigen?

**Comment by nidegen** [comment]
i hoped cmake find would work, as it is system independent.

will figure out how to do it

**Pull request merged**